wayback/_client.py

```diff
@@ -226,7 +226,7 @@ class WaybackSession(_utils.DisableAfterCloseSession, requests.Session):
     # just the error type. See `should_retry_error()`.
     handleable_errors = (ConnectionError,) + retryable_errors
 
-    def __init__(self, retries=6, backoff=2, timeout=None, user_agent=None):
+    def __init__(self, retries=6, backoff=2, timeout=60, user_agent=None):
```

Comment:
Ah, this is actually really long! (Have you encountered many Wayback cases where this is what you need?)

This isn’t a timeout for the whole request/response cycle, but a timeout between reads on the network socket — i.e. this is saying if you go more than N seconds between receiving packets from the Wayback Machine servers, then cancel. Usually this is more like 2 (short, but common) to 20 (pretty long for most use-cases). Since Wayback is definitely a little slow, I think something more on the long end is good, but 60 seems pretty far out there to me. 😳

Reply:
Actually, maybe it’s fine that this is long. It’s at least some outside bound! Nevermind my earlier comment.

Reply:
Right, the timeout is accessible from within the package. What I meant is that a user cannot provide a custom timeout when calling `WaybackClient.search`. But I don't know if this is even necessary.

Yeah, 60 seconds is extremely long. But I've had problems with my script completely freezing every couple hundred searches. So this timeout would prevent the me from waiting days for a response without the chance of stopping an request that is actually getting answered.

Reply:
> What I meant is that a user cannot provide a custom timeout when calling `WaybackClient.search`. But I don't know if this is even necessary.

Ah, I see! I don’t think we need to change that in this PR (the current path for doing that is creating a custom session when you instantiate the client), but it would be great to add `timeout` as a keyword argument to both `search()` and `get_memento()` as a follow-on to this. Can you make an issue for that?

> I've had problems with my script completely freezing every couple hundred searches. So this timeout would prevent the me from waiting days for a response without the chance of stopping an request that is actually getting answered.

👍 👍 👍

Reply:
OK